samplejava/src/main/java/com/example/chattutorialjava/ImgurAttachmentViewHolder.java

```diff
@@ -0,0 +1,95 @@
+package com.example.chattutorialjava;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+
+import androidx.constraintlayout.widget.ConstraintLayout;
+import androidx.constraintlayout.widget.ConstraintSet;
+
+import com.example.chattutorialjava.databinding.ViewHolderImgurAttachmentBinding;
+import com.getstream.sdk.chat.adapter.MessageListItem;
+import com.google.android.material.shape.ShapeAppearanceModel;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+import coil.Coil;
+import coil.request.ImageRequest;
+import io.getstream.chat.android.client.models.Attachment;
+import io.getstream.chat.android.ui.messages.adapter.BaseMessageItemViewHolder;
+import io.getstream.chat.android.ui.messages.adapter.MessageListItemPayloadDiff;
+
+class ImgurAttachmentViewHolder extends BaseMessageItemViewHolder<MessageListItem.MessageItem> {
+
+    ViewHolderImgurAttachmentBinding binding;
+
+    public static ImgurAttachmentViewHolder create(ViewGroup parent) {
+        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+        ViewHolderImgurAttachmentBinding binding = ViewHolderImgurAttachmentBinding.inflate(inflater, parent, false);
+        return new ImgurAttachmentViewHolder(binding);
+    }
+
+    private ImgurAttachmentViewHolder(@NotNull ViewHolderImgurAttachmentBinding binding) {
+        super(binding.getRoot());
+
+        float cornerRadius = binding.getRoot()
+                .getResources()
+                .getDimension(R.dimen.stream_ui_selected_attachment_corner_radius);
+
+        ShapeAppearanceModel model = binding.ivMediaThumb.getShapeAppearanceModel()
+                .toBuilder()
+                .setAllCornerSizes(cornerRadius)
+                .build();
+
+        binding.ivMediaThumb.setShapeAppearanceModel(model);
+        binding.ivMediaThumb.setScaleType(ImageView.ScaleType.CENTER_CROP);
+
+        this.binding = binding;
+    }
+
+    @Override
+    public void bindData(
+            @NotNull MessageListItem.MessageItem data,
+            @Nullable MessageListItemPayloadDiff diff
+    ) {
+        List<Attachment> attachments = data.getMessage().getAttachments();
+        String imageUrl = attachments.get(0).getImageUrl();
+        Context context = getContext();
```

Comment:
`getContext()` doesn't exist on the new base ViewHolder, only as an extension. We can probably just grab the `itemView` and its context here instead.

Reply:
This is one of the changes that depends on the next SDK release. It's implemented by retrieving context from the `itemView`.